core/integration/module_test.go

```diff
@@ -3442,6 +3442,57 @@ class Test:
     @function
     async def gimme_dir_ents(self) -> list[str]:
         return await self.dir.entries()
+`,
+			},
+			{
+				sdk: "typescript",
+				source: `
+import { Directory, object, func, field } from '@dagger.io/dagger';
+
+@object
+class Test {
+	@field
+	foo: string
+				
+	@field
+	dir: Directory
+				
+	@field
+	bar: number
+				
+	@field
+	baz: string[]
+				
+	@field
+	neverSetDir?: Directory = undefined
```

Comment:
Just curious, does this work?
```suggestion
	neverSetDir?: Directory
```

Reply:
It does! I'll update that after your review